### PR TITLE
Windows: fix tests to work with test wrapper

### PR DESCRIPTION
**Pull request by laszlocsomor**
Fix the shell tests to work with
--experimental_windows_native_test_wrapper (when
using Bazel built at HEAD).

In this change:

- Remove the is_windows function definition from
  testenv.sh, to ensure that tests wanting to
  check the platform will define their own
  is_windows variable (as many tests already do).

- Define the is_windows variable using `uname` in
  every test that needs to know the platform.

- Use the Bash runfiles library in the affected
  tests.

- Every test that calls "bazel shutdown" now uses
  it with try_with_timeout, so tests won't fail
  flakily when bazel cannot immediately be shut
  down.

See https://github.com/bazelbuild/bazel/issues/5508

### CODE REVIEW

File: src/test/shell/testenv.sh [hunk 1]
```diff
@@ -19,12 +19,22 @@
...
 
-PLATFORM="$(uname -s | tr 'A-Z' 'a-z')"
-
-function is_windows() {
-  # On windows, the shell test is actually running on msys
-  [[ "${PLATFORM}" =~ msys_nt* ]]
-}
+PLATFORM="$(uname -s | tr [:upper:] [:lower:])"
+
+# # `uname` returns the current platform, e.g "MSYS_NT-10.0" or "Linux".
```
Comment (meteorcloudy) [inline src/test/shell/testenv.sh]: Are those unnecessary anymore? Could we just remove them?

File: src/test/shell/integration/execution_phase_tests.sh [hunk 2]
```diff
@@ -64,7 +64,7 @@ function set_up() {
 }
 
 function tear_down() {
-    bazel shutdown
+  try_with_timeout bazel shutdown
```
Comment (meteorcloudy) [inline src/test/shell/integration/execution_phase_tests.sh]: What is the timeout? Is it enough for `bazel shutdown`?

Comment (laszlocsomor) [reply src/test/shell/integration/execution_phase_tests.sh]: 120 seconds. This is defined in testenv.sh. It has been enough for me so far.

File: src/test/shell/testenv.sh (hunk 1 above)
Comment (laszlocsomor) [reply src/test/shell/testenv.sh]: Done.

File: src/test/shell/integration/execution_phase_tests.sh (hunk 2 above)
Comment (meteorcloudy) [reply src/test/shell/integration/execution_phase_tests.sh]: Got it, thanks!

**Pull request closed without merging**